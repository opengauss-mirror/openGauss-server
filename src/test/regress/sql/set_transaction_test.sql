set b_format_behavior_compat_string = '';
\h SET TRANSACTION
SET GLOBAL TRANSACTION READ ONLY;
SET SESSION TRANSACTION READ ONLY;
create database test_set_tran dbcompatibility 'b';
\c test_set_tran
SET SESSION TRANSACTION READ ONLY;
CREATE DATABASE test_set_tran1;
set b_format_behavior_compat_options = 'set_session_transaction';
SET SESSION TRANSACTION READ ONLY;
CREATE DATABASE test_set_tran2;
\c test_set_tran;
CREATE DATABASE test_set_tran2;
\c test_set_tran;
SET GLOBAL TRANSACTION READ ONLY;
CREATE DATABASE test_set_tran3;
\c test_set_tran;
CREATE DATABASE test_set_tran4;
SET GLOBAL TRANSACTION READ WRITE;
CREATE DATABASE test_set_tran4;
\c test_set_tran;
CREATE DATABASE test_set_tran4;
SET GLOBAL TRANSACTION ISOLATION LEVEL SERIALIZABLE;
SET GLOBAL TRANSACTION ISOLATION LEVEL SERIALIZABLE READ ONLY;
SET GLOBAL TRANSACTION ISOLATION LEVEL SERIALIZABLE READ WRITE;
SET GLOBAL TRANSACTION ISOLATION LEVEL REPEATABLE READ;
SET GLOBAL TRANSACTION ISOLATION LEVEL REPEATABLE READ READ ONLY;
SET GLOBAL TRANSACTION ISOLATION LEVEL REPEATABLE READ READ WRITE;
SET GLOBAL TRANSACTION ISOLATION LEVEL READ UNCOMMITTED;
SET GLOBAL TRANSACTION ISOLATION LEVEL READ UNCOMMITTED READ ONLY;
SET GLOBAL TRANSACTION ISOLATION LEVEL READ UNCOMMITTED READ WRITE;
SET GLOBAL TRANSACTION ISOLATION LEVEL READ COMMITTED;
SET GLOBAL TRANSACTION ISOLATION LEVEL READ COMMITTED READ ONLY;
SET GLOBAL TRANSACTION ISOLATION LEVEL READ COMMITTED READ WRITE;
\c test_set_tran;
SET SESSION TRANSACTION ISOLATION LEVEL SERIALIZABLE;
SET SESSION TRANSACTION ISOLATION LEVEL SERIALIZABLE READ ONLY;
SET SESSION TRANSACTION ISOLATION LEVEL SERIALIZABLE READ WRITE;
SET SESSION TRANSACTION ISOLATION LEVEL REPEATABLE READ;
SET SESSION TRANSACTION ISOLATION LEVEL REPEATABLE READ READ ONLY;
SET SESSION TRANSACTION ISOLATION LEVEL REPEATABLE READ READ WRITE;
SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED;
SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED READ ONLY;
SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED READ WRITE;
SET SESSION TRANSACTION ISOLATION LEVEL READ COMMITTED;
SET SESSION TRANSACTION ISOLATION LEVEL READ COMMITTED READ ONLY;
SET SESSION TRANSACTION ISOLATION LEVEL READ COMMITTED READ WRITE;
\c test_set_tran;
CREATE USER newuser PASSWORD 'gauss@123';
SET SESSION AUTHORIZATION newuser PASSWORD 'gauss@123';
SET GLOBAL TRANSACTION ISOLATION LEVEL READ COMMITTED READ ONLY;
RESET SESSION AUTHORIZATION;
\c regression;
DROP DATABASE test_set_tran;
DROP DATABASE test_set_tran1;
DROP DATABASE test_set_tran2;
DROP DATABASE test_set_tran3;
DROP DATABASE test_set_tran4;