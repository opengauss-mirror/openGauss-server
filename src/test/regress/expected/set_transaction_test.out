set b_format_behavior_compat_string = '';
\h SET TRANSACTION
Command:     SET TRANSACTION
Description: set the characteristics of the current transaction
Syntax:
{SET [ LOCAL | SESSION | GLOBAL ] TRANSACTION|SET SESSION CHARACTERISTICS AS TRANSACTION}
  { ISOLATION LEVEL { READ COMMITTED | READ UNCOMMITTED }
  | { READ WRITE | READ ONLY | SERIALIZABLE | REPEATABLE READ }
  } [, ...]
SET TRANSACTION SNAPSHOT snapshot_id;
NOTICE: SET GLOBAL TRANSACTION is only available in CENTRALIZED mode and B-format database!

SET GLOBAL TRANSACTION READ ONLY;
ERROR:  SET GLOBAL TRANSACTION is only supported in B_FORMAT.
SET SESSION TRANSACTION READ ONLY;
create database test_set_tran dbcompatibility 'b';
\c test_set_tran
SET SESSION TRANSACTION READ ONLY;
CREATE DATABASE test_set_tran1;
set b_format_behavior_compat_options = 'set_session_transaction';
SET SESSION TRANSACTION READ ONLY;
CREATE DATABASE test_set_tran2;
ERROR:  cannot execute CREATE DATABASE in a read-only transaction
\c test_set_tran;
CREATE DATABASE test_set_tran2;
\c test_set_tran;
SET GLOBAL TRANSACTION READ ONLY;
CREATE DATABASE test_set_tran3;
\c test_set_tran;
CREATE DATABASE test_set_tran4;
ERROR:  cannot execute CREATE DATABASE in a read-only transaction
SET GLOBAL TRANSACTION READ WRITE;
CREATE DATABASE test_set_tran4;
ERROR:  cannot execute CREATE DATABASE in a read-only transaction
\c test_set_tran;
CREATE DATABASE test_set_tran4;
SET GLOBAL TRANSACTION ISOLATION LEVEL SERIALIZABLE;
SET GLOBAL TRANSACTION ISOLATION LEVEL SERIALIZABLE READ ONLY;
SET GLOBAL TRANSACTION ISOLATION LEVEL SERIALIZABLE READ WRITE;
SET GLOBAL TRANSACTION ISOLATION LEVEL REPEATABLE READ;
SET GLOBAL TRANSACTION ISOLATION LEVEL REPEATABLE READ READ ONLY;
SET GLOBAL TRANSACTION ISOLATION LEVEL REPEATABLE READ READ WRITE;
SET GLOBAL TRANSACTION ISOLATION LEVEL READ UNCOMMITTED;
SET GLOBAL TRANSACTION ISOLATION LEVEL READ UNCOMMITTED READ ONLY;
SET GLOBAL TRANSACTION ISOLATION LEVEL READ UNCOMMITTED READ WRITE;
SET GLOBAL TRANSACTION ISOLATION LEVEL READ COMMITTED;
SET GLOBAL TRANSACTION ISOLATION LEVEL READ COMMITTED READ ONLY;
SET GLOBAL TRANSACTION ISOLATION LEVEL READ COMMITTED READ WRITE;
\c test_set_tran;
SET SESSION TRANSACTION ISOLATION LEVEL SERIALIZABLE;
SET SESSION TRANSACTION ISOLATION LEVEL SERIALIZABLE READ ONLY;
SET SESSION TRANSACTION ISOLATION LEVEL SERIALIZABLE READ WRITE;
SET SESSION TRANSACTION ISOLATION LEVEL REPEATABLE READ;
SET SESSION TRANSACTION ISOLATION LEVEL REPEATABLE READ READ ONLY;
SET SESSION TRANSACTION ISOLATION LEVEL REPEATABLE READ READ WRITE;
SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED;
SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED READ ONLY;
SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED READ WRITE;
SET SESSION TRANSACTION ISOLATION LEVEL READ COMMITTED;
SET SESSION TRANSACTION ISOLATION LEVEL READ COMMITTED READ ONLY;
SET SESSION TRANSACTION ISOLATION LEVEL READ COMMITTED READ WRITE;
\c test_set_tran;
CREATE USER newuser PASSWORD 'gauss@123';
SET SESSION AUTHORIZATION newuser PASSWORD 'gauss@123';
SET GLOBAL TRANSACTION ISOLATION LEVEL READ COMMITTED READ ONLY;
ERROR:  permission denied for database test_set_tran
DETAIL:  N/A
RESET SESSION AUTHORIZATION;
\c regression;
DROP DATABASE test_set_tran;
DROP DATABASE test_set_tran1;
DROP DATABASE test_set_tran2;
DROP DATABASE test_set_tran3;
DROP DATABASE test_set_tran4;
